File: Source/Aladdin.CAPI.ANSI.Java/src/aladdin/asn1/ansi/x962/SpecifiedMultiple.java
package aladdin.asn1.ansi.x962;
import aladdin.asn1.*; 
import aladdin.asn1.Integer; 
import java.io.*; 

////////////////////////////////////////////////////////////////////////////////
// SpecifiedMultiples ::= SEQUENCE {
//      multiple INTEGER,
//      point ECPoint 
// }
////////////////////////////////////////////////////////////////////////////////
public final class SpecifiedMultiple extends Sequence<IEncodable>
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(Integer    .class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(OctetString.class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public SpecifiedMultiple(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public SpecifiedMultiple(Integer multiple, OctetString point) { super(info, multiple, point); }

	public final Integer     multiple() { return (Integer    )get(0); }
	public final OctetString point   () { return (OctetString)get(1); }
}

File: Source/Aladdin.CAPI.ANSI/ASN1/ANSI/X962/SpecifiedMultiple.cs
using System; 
using System.Runtime.Serialization;

namespace Aladdin.ASN1.ANSI.X962
{
    ////////////////////////////////////////////////////////////////////////////////
    // SpecifiedMultiples ::= SEQUENCE {
    //      multiple INTEGER,
    //      point ECPoint 
    // }
    ////////////////////////////////////////////////////////////////////////////////
	[Serializable]
    public class SpecifiedMultiple : Sequence
    {
	    // информация о структуре
	    private static readonly ObjectInfo[] info = new ObjectInfo[] { 

		    new ObjectInfo(new ObjectCreator<Integer    >().Factory(), Cast.N), 
		    new ObjectInfo(new ObjectCreator<OctetString>().Factory(), Cast.N), 
	    }; 
		// конструктор при сериализации
        protected SpecifiedMultiple(SerializationInfo info, StreamingContext context) : base(info, context) {}

	    // конструктор при раскодировании
	    public SpecifiedMultiple(IEncodable encodable) : base(encodable, info) {}

	    // конструктор при закодировании
	    public SpecifiedMultiple(Integer multiple, OctetString point) : base(info, multiple, point) {}

	    public Integer     Multiple { get { return (Integer    )this[0]; }}
	    public OctetString Point    { get { return (OctetString)this[1]; }}
    }
}